### PR TITLE
Creating compat layers for Button, Checkbox, Link, Pivot, Slider and Toggle components 

### PR DESCRIPTION
**Pull request by khmakoto**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ yarn change`

#### Description of changes

This PR replaces the `Button` exported in `react-next` that was pointing to the one in `office-ui-fabric-react` to the one that is pointing to `react-button`, while creating a compatibility layer in `react-next` to still export the `office-ui-fabric-react`.

It also creates this compatibility layer for the `Checkbox`, `Link`, `Pivot`, `Slider` and `Toggle` components to have separate exports for the `office-ui-fabric-react` and `@fluentui/react-next` components.

Finally, this PR makes changes to allow for multiple `api-extractor` configs to be provided per package and makes changes to `lint-imports` so compat imports are be flagged by the deep imports rule.

#### Focus areas to test

(optional)

### CODE REVIEW

**Comment by xugao** [comment]
this LGTM , but i will let @joschect to approve :)

File: packages/react-next/etc/react-next.api.md [hunk 1]
```diff
@@ -5,9 +5,9 @@
 ```ts
 
 import { ComposePreparedOptions } from '@fluentui/react-compose';
-import { IButtonProps } from 'office-ui-fabric-react/lib/Button';
+import { IButtonProps } from '@fluentui/react-next/lib/CompatButton';
```
Comment (JustSlone) [inline packages/react-next/etc/react-next.api.md:8]: Should this be something more like `'@fluentui/react-next/lib/compat/Button'`

Comment (khmakoto) [reply packages/react-next/etc/react-next.api.md:8]: I thought we didn't want to allow for imports from internal folders so we restricted that to imports from the overall package, i.e. `@fluentui/react-next` and imports from specific files under `lib`, i.e. `@fluentui/react-next/lib/<component>`.

File: packages/react-next/etc/react-next.api.md [hunk 2]
```diff
@@ -4,12 +4,66 @@
...
+import { CompoundButton } from 'office-ui-fabric-react/lib/Button';
+import { DefaultButton } from 'office-ui-fabric-react/lib/Button';
+import { ElementType } from 'office-ui-fabric-react/lib/Button';
+import { IBaseButtonProps } from 'office-ui-fabric-react/lib/Button';
+import { IBaseButtonState } from 'office-ui-fabric-react/lib/Button';
 import { IBaseProps } from 'office-ui-fabric-react/lib/Utilities';
+import { IButton } from 'office-ui-fabric-react/lib/Button';
 import { IButtonProps } from 'office-ui-fabric-react/lib/Button';
-import { IButtonProps as IButtonProps_2 } from 'office-ui-fabric-react/lib/components/Button/Button.types';
+import { IButtonProps as IButtonProps_2 } from '@fluentui/react-next/lib/CompatButton';
```
Comment (ecraig12345) [inline packages/react-next/etc/react-next.api.md:35]: Anywhere you're getting a _2 or _3 means that you're exporting the same name twice with different definitions, which is going to cause problems for consumers.

File: packages/react-next/etc/react-next.api.md (hunk 1 above)
Comment (ecraig12345) [reply packages/react-next/etc/react-next.api.md]: I would prefer `compat/Button`, and we could update our guidance to indicate paths immediately under `compat` are also considered top-level paths that are part of the public API.

Another nice thing about that approach is we could have a file `compat/index` where you could get any compat component exported by its original name. Then only do the renaming when re-exporting from the root (or maybe even better, we don't re-export those things from the root at all).

Comment (dzearing) [reply packages/react-next/etc/react-next.api.md]: `@fluentui/react-next/lib/compat`

Comment (ecraig12345) [reply packages/react-next/etc/react-next.api.md]: That's what I meant as well, I was just being lazy with typing 😆

File: packages/react-next/src/CompatCheckbox.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export {
+  Checkbox as CompatCheckbox,
```
Comment (dzearing) [inline packages/react-next/src/CompatCheckbox.ts:2]: src/compat/Checkbox.ts
src/compat/index.ts // (exports * from checkbox)

Comment (dzearing) [reply packages/react-next/src/CompatCheckbox.ts:2]: Then someone importing:

/lib/Button => /lib/compat/Button
/lib/Checkbox => /lib/compat/Checkbox

File: packages/react-next/etc/react-next.api.md (hunk 1 above)
Comment (ecraig12345) [reply packages/react-next/etc/react-next.api.md]: Err to clarify, what I actually meant was something like this, basically mirroring our existing structure that lets you import from either the index (all components) or individual component files:
```
@fluentui/react-next/lib/compat/
  index
  Button
  Checkbox
  ...any other component with compat layer...
```

File: packages/react-next/src/CompatCheckbox.ts (hunk 3 above)
Comment (khmakoto) [reply packages/react-next/src/CompatCheckbox.ts]: Done

File: packages/react-next/etc/react-next.api.md (hunk 2 above)
Comment (khmakoto) [reply packages/react-next/etc/react-next.api.md]: Resolved!

File: packages/react-next/etc/react-next.api.md (hunk 1 above)
Comment (khmakoto) [reply packages/react-next/etc/react-next.api.md]: Cool, I went with this!

File: packages/react-next/package.json [hunk 4]
```diff
@@ -30,7 +30,7 @@
     "start": "just-scripts dev:storybook",
     "start-test": "just-scripts jest-watch",
     "update-snapshots": "just-scripts jest -u",
-    "update-api": "just-scripts update-api"
+    "update-api": "just-scripts update-api && just-scripts update-api --configPath ./config/api-extractor-compat.json"
```
Comment (xugao) [inline packages/react-next/package.json:33]: we should add `api-extractor:verify ./config/api-extractor-compat.json` to `build` too?

**Comment by xugao** [comment]
pls pull master and get a good `size-auditor` run

**Pull request merged**